pkg/ui/ccl/src/views/clusterviz/containers/map/currentNodes.tsx

```diff
@@ -0,0 +1,254 @@
+// Copyright 2017 The Cockroach Authors.
+//
+// Licensed under the Cockroach Community Licence (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/LICENSE
+
+import _ from "lodash";
+import React from "react";
+import { connect } from "react-redux";
+import * as d3 from "d3";
+
+import { refreshNodes, refreshLiveness } from "src/redux/apiReducers";
+import { nodesSummarySelector, NodesSummary } from "src/redux/nodes";
+import { AdminUIState } from "src/redux/state";
+
+import NodeStatusHistory from "./statusHistory";
+import * as Vector from "./vector";
+import * as PathMath from "./pathmath";
+import { Locality } from "./locality";
+import { LocalityLink, LocalityLinkProps } from "./localitylink";
+
+export interface CurrentNodesProps {
+  nodesSummary: NodesSummary;
+  statusesValid: boolean;
+  refreshNodes: typeof refreshNodes;
+  refreshLiveness: typeof refreshLiveness;
+}
+
+export interface CurrentNodesOwnProps {
+  projection: d3.geo.Projection;
+}
+
+// List of fake location data in order to place nodes on the map for visual
+// effect.
+const locations: [number, number][] = [
+  [-74.00597, 40.71427],
+  [-80.19366, 25.77427],
+  [-93.60911, 41.60054],
+  [-118.24368, 34.05223],
+  [-122.33207, 47.60621],
+  [-0.12574, 51.50853],
+  [13.41053, 52.52437],
+  [18.0649, 59.33258],
+  [151.20732, -33.86785],
+  [144.96332, -37.814],
+  [153.02809, -27.46794],
+  [116.39723, 39.9075],
+  [121.45806, 31.22222],
+  [114.0683, 22.54554],
+  [72.88261, 19.07283],
+  [77.59369, 12.97194],
+  [77.22445, 28.63576],
+];
+
+// CurrentNodes places all current nodes on the map based on their long/lat
+// coordinates (currently simulated). Also generates a link between each node
+// pair, visualized on the map as a directed path.
+class CurrentNodes extends React.Component<CurrentNodesProps & CurrentNodesOwnProps, any> {
```

Comment:
Is the plan to replace this with `CurrentLocalities` in order to deal with the aggregations?